### PR TITLE
[haproxy] provide options for collapsing `count_per_status` metrics

### PR DESCRIPTION
**Pull request by talwai**
Adds two flags to the `haproxy` config:
   - `count_status_by_service` will instruct the check to tag the status counts it collects by service in addition to by host. Defaults `True`
   - `collate_status_tags_per_host` will instruct the check to collate backend-level `status` tags for `haproxy.count_per_status` from `('up', 'open', 'down', 'maint', 'nolb')` to one of `(up, down)`

### CODE REVIEW

File: checks.d/haproxy.py [hunk 1]
```diff
@@ -309,14 +334,15 @@ def _process_status_metric(self, hosts_statuses, collect_status_metrics_by_host,
...
+                tags = ['status:%s' % status]
+                if aggregate_status_by_service:
+                    tags.append('service:%s' % service)
+
                 self.gauge("haproxy.count_per_status", count, tags=tags)
 
     def _gauge_all_statuses(self, metric_name, count, status, tags):
-        self.gauge(metric_name, count, tags + ['status:%s' % status])
-        for state in Services.ALL_STATUSES:
-            if state != status:
```
Comment (truthbk) [inline checks.d/haproxy.py]: thanks for the clanup here. :clap:

File: conf.d/haproxy.yaml.example [hunk 2]
```diff
@@ -25,6 +25,16 @@ instances:
...
+    # The (optional) collect_count_per_status parameter will instruct
+    # the check to collect counts for each status by host.
+    # This only applies if collect_status_metrics is True.
+    # collect_count_per_status: True
+    #
+    # The (optional) aggregate_status_by_service parameter will instruct
+    # the check to tag the status counts it collects by service in addition to by host.
+    # This only applies if collect_status_metrics and collect_count_per_status are True.
+    # aggregate_status_by_service: True
+    #
```
Comment (truthbk) [inline conf.d/haproxy.yaml.example]: Defaulting to True, so backward-compatible. :+1:

**Comment by truthbk** [comment]
Looks good to me! :+1:

**Comment by talwai** [comment]
```
Defaults   - Collected 174 metrics, 0 events & 20 service checks 
collate_status_tags_per_host: true - Collected 138 metrics, 0 events & 32 service checks
count_status_by_service: false - Collected 104 metrics, 0 events & 32 service checks
```

**Comment by yannmh** [comment]
Closing in favor of https://github.com/DataDog/dd-agent/pull/2304

**Pull request closed without merging**